src/main/java/io/cdap/delta/datastream/DatastreamEventReader.java

```diff
@@ -107,7 +112,14 @@ public DatastreamEventReader(DatastreamConfig config, EventReaderDefinition defi
     this.config = config;
     this.definition = definition;
     this.emitter = emitter;
-    this.executorService = Executors.newSingleThreadScheduledExecutor();
+
+    this.bucketCreatedByCDF = new String(context.getState("bucketCreatedByCDF"), StandardCharsets.UTF_8).equals("true");
```

Comment:
getState returns an empty byte array if the key doesn't exist. The IDE suggests adding Objects.requireNonNull around the function call, is that sufficient?

Reply:
Objects.requireNonNull would just throw NullPointerException if obj is null, although `context.getState` returns an empty array today, but it would be better to not rely on it and gracefully handle null scenario

Reply:
Alright got it